File: sundb/src/test/java/com/dbx/agent/sundb/SundbAgentMetadataTest.java
package com.dbx.agent.sundb;

import com.dbx.agent.test.TestSupport;
import com.dbx.agent.test.JdbcAgentFake;
import com.dbx.agent.test.JdbcMetadataSqlFake;
import java.util.Collections;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class SundbAgentMetadataTest {
    @Test
    void quotesSchemaAndTableIdentifiersInIndexMetadataSql() {
        SundbAgent agent = new SundbAgent();
        TestSupport.setPrivateConnection(agent, JdbcMetadataSqlFake.connection());

        agent.listIndexes("bad`schema", "bad`table");

        Assertions.assertEquals(
            Collections.singletonList("SHOW INDEX FROM `bad``table` FROM `bad``schema`"),
            JdbcMetadataSqlFake.statements
        );
    }
}

File: sundb/src/test/kotlin/com/dbx/agent/sundb/SundbAgentMetadataTest.kt
package com.dbx.agent.sundb

import com.dbx.agent.test.JdbcMetadataSqlFake
import com.dbx.agent.test.setPrivateConnection
import kotlin.test.Test
import kotlin.test.assertEquals

class SundbAgentMetadataTest {
    @Test
    fun `quotes schema and table identifiers in index metadata SQL`() {
        val agent = SundbAgent()
        val fake = JdbcMetadataSqlFake.connection()
        setPrivateConnection(agent, fake)

        agent.listIndexes("bad`schema", "bad`table")

        assertEquals(
            listOf("SHOW INDEX FROM `bad``table` FROM `bad``schema`"),
            JdbcMetadataSqlFake.statements,
        )
    }
}
